### PR TITLE
Replace __unicode__ methods with __str__ and modernize some string formatting

### LINKED ISSUE #844: Fix model string representations
## General Checkups
- [x] Have you checked that there isn't already an existing issue that describes what you report below?
- [x] Have you checked that there isn't already an open pull requests for this issue/update/change?

## Description
Models have stopped displaying string representations as expected; see email to support. I think it might have to do with this -- seems to indicate that the `__unicode__` methods are obsolete/nonfunctional in Python 3? https://docs.djangoproject.com/en/1.11/topics/python3/#str-and-unicode-methods

## Example
The email to support mentioned above contained: `your account at <open_humans.settings.FakeSite object at 0x7f386bda6240>.` when it should contain the `name`.

### PR DESCRIPTION
**Pull request by jhdulaney**
## General Checkups
- [*] Have you checked that there aren't other open pull requests for the same issue/update/change?
- [*] If your code includes new features and not just bug fixes/copy editing: Did you include new tests?

## Description
During the Python3 migration, I neglected to replace several __unicode__ methods with __str__ methods.  This has been done.  In addition, I modernized a few of the instances of string formatting.

## Related Issue
Fixes #844 

Signed-off-by: Mairi Dulaney <jdulaney@fedoraproject.org>

### CODE REVIEW

File: open_humans/models.py [hunk 1]
```diff
@@ -26,21 +26,21 @@ def get_member_profile_image_upload_path(instance, filename):
...
+    return str('grant-projects/{0}/{1}').format(instance.name, filename)
 
 def random_member_id():
     """
     Return a zero-padded string from 00000000 to 99999999 that's not in use by
     any Member.
     """
     def random_id():
-        return '%08d' % random.randint(0, 99999999)
+        return str('{0:d}').format(random.randint(0, 99999999))
```
Comment (gedankenstuecke) [inline open_humans/models.py]: Are you sure this produces the right kind of padding? 

When running in an ipython shell I ended up with: 

```
In [52]: str('{0:d}').format(random.randint(0,99999999))
Out[52]: '91987'
```

File: open_humans/models.py [hunk 2]
```diff
@@ -26,21 +26,21 @@ def get_member_profile_image_upload_path(instance, filename):
     """
     Construct the upload path for a given member and filename.
     """
-    return 'member/%s/profile-images/%s' % (instance.user.id, filename)
+    return str('member/{0}/profile-images/{1}').format(instance.user.id, filename)
```
Comment (gedankenstuecke) [inline open_humans/models.py]: I noticed that we're giving the `{0}/{1}.format(foo,bar)` formatting in some places and in others only the `{}/{}` notation. Is there a reason why?

File: open_humans/models.py (hunk 1 above)
Comment (jhdulaney) [reply open_humans/models.py]: Ah, oops

File: open_humans/models.py (hunk 2 above)
Comment (jhdulaney) [reply open_humans/models.py]: No real reason.  I'll make them consistent.

Comment (gedankenstuecke) [reply open_humans/models.py]: nice, thanks!

File: public_data/models.py [hunk 3]
```diff
@@ -77,8 +77,8 @@ def __str__(self):
         if self.is_public:
             status = 'Public'
 
-        return str('{}:{}:{}').format(self.participant.member.user.username,
-                             self.data_source, status)
+        return str('{0}:{1}:{1}').format(self.participant.member.user.username,
```
Comment (gedankenstuecke) [inline public_data/models.py]: this should be `{0}:{1}:{2}` i think 😉

**Pull request merged**